### DISCUSSION TITLE
Bitwarden Android issues

### DISCUSSION THREAD

**Discussion by davidfrickert** [dani-garcia/vaultwarden#5152] [Q&A]
So apparently bitwarden android got a big update yesterday and at least for me it doesn't work with vaultwarden (tried clean install, clear cache, etc).

Can log-in without issues but then when trying to get passwords there is an error.

Anyone else experiencing this?

Downgraded to v 2024.10.0 for now and that one seems to work fine.

**Comment by BlackDex** [comment]
Please verify you have installed the very latest (and greatest) Vaultwarden version v1.32.3, if not you will not be able to login!

**Reply by davidfrickert** [reply]
I do have it!

**Reply by davidfrickert** [reply]
1.32.3-alpine

**Comment by davidfrickert** [comment]
okay... it seems to work now, i just restarted vaultwarden and restored the new android version and now it works... idk.

**Reply by davidfrickert** [reply]
actually okay i do know what's wrong.
i am running an HA deployment of vaultwarden (3 instances), it seems that with this update this no longer works :cry: 
I reverted back to one instance and the app works fine

**Reply by BlackDex** [reply]
Nothing changed which could affect that. Though we do not support HA specifically. It should just work fine.
Check the logs, and be sure the rsa key is exactly the same on all those pods/instances.

**Reply by davidfrickert** [reply]
hmm yeah i will have a look at the logs, i believe the issue would be on the updated android app and not on vaulwarden but i'm not sure.
and it is as you say, HA is technically not supported so yeah

**Comment by jiulingz** [comment]
I am experiencing the same issue. Tried the mentioned actions above and the app still doesn't work

Vaultwarden Server
- Version: 1.32.3
- Restarted and there's only one docker instance
- Log when I press "Sync" or "Try again" on the app
```
vault-warden  | [2024-11-02 06:57:44.102][request][INFO] GET /api/accounts/revision-date
vault-warden  | [2024-11-02 06:57:44.103][response][INFO] (revision_date) GET /api/accounts/revision-date => 200 OK
vault-warden  | [2024-11-02 06:57:44.114][request][INFO] GET /api/sync
vault-warden  | [2024-11-02 06:57:44.135][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
```

Bitwarden android app 
- Version: v2024.10.2 from Google Play Store
- Tried clear cache, clear storage, clean install
- Error message is shown for the "My vault" and "Send" tab, with a "Try again" button
```
We are unable to process your request. Please try again or contact us.
```
![Send](https://github.com/user-attachments/assets/b5c422ab-3e97-415c-b672-aee5f4410698)
![MyVault](https://github.com/user-attachments/assets/c35c0bc5-a3a7-40d4-b4f7-fb47b8ed77cc)

**Comment by FlaxHaxx** [comment]
I had the same issue after the Android app update. It works again after updating Vaultwarden to version 1.32.3 and reinstalling the Android app.

**Comment by BlackDex** [comment]
@jiulingz Do you have it working already?
You have verified Vaultwarden is the latest version right?
And you tried fully uninstall and re-install the client app on Android right?
And you can see/access all ciphers in the web-vault?

If all of the above is correct. I would like to suggest to try the following.
**But first! Create a backup!!**

1. Login in into the web-vault.
2. Go to your account settings
3. Select the **Master Password** tab
4. Check the **Also rotate my account's encryption key** checkbox and confirm the popup
5. Provide your password. You can keep you password exactly the same, just enter your password 3 times
6. Click on the **Change master password** button and wait a while.

It could take a while for this to finish, and it could also generate a (gateway) timeout error, but that shouldn't be a problem.
You can check the Vaultwarden logs to see if anything went wrong or not.

After all that, try to login again and see if you still have access to everything.
If that is the case, try to login using your mobile client and see if that solved the issue.

![image](https://github.com/user-attachments/assets/f43fdcbb-3eba-4e9f-ab9f-d12b4b1f7509)

**Reply by jiulingz** [reply]
Thanks for the instructions. Unfortunately the Android app still doesn't work after I rotated the encryption key. I am still receiving the same error. 

- Vaultwarden is the latest version 1.32.3.
- Vaultwarden logs didn't show any error while rotating the encryption key.
- I have fully uninstalled and reinstalled the Android app, after rotating the encryption key as well.
- I have restarted the Vaultwarden server (docker) after rotating the encryption key.
- I am able to access all ciphers in the web vault, Firefox add-on, and rbx. The only problematic client is the Android app.

**Reply by BlackDex** [reply]
Then it must be something very strange, or linked to an org cipher. The only way to rule that it is to remove your self from the orgs you are a member from. And if that works, then it's a vault item in that org.

**Reply by jiulingz** [reply]
Yes it was a problematic entry in the organization. The entry itself has long notes and an attachment. 

Not sure about the root cause, or the simplest solution, but the problem was fixed after me deleting the attachment, cloning the item, removing the original, re-adding the attachment to the clone.

**Reply by BlackDex** [reply]
Any idea which client (and maybe version or date added) was used to add this item? 

It might help a bit in finding a solution.

**Reply by jiulingz** [reply]
It was added a long time ago (maybe 6m to 1yr ago?), mostly likely through the Firefox add-on, but I can't say for sure.

**Reply by eltroniq** [reply]
Hi @BlackDex , I followed all above suggestions, unfortunately it still does not work for me. I also removed myself from all organizations. Multiple reinstalls of the app (also including different versions) and cleaning data didn't work, as well as rebooting server and mobile. Also http/https URLs doesn't seem to make any difference in the app.
I am running the docker image vaultwarden/server 1.32.4. Android app via F-Droid from [bitwarden](https://mobileapp.bitwarden.com/fdroid/repo/bitwarden_v2024.11.3.apk).

Server logs only show something right after setting the URLs in the app. After that nothing.
I am pretty clueless right now, any help would be greatly appreciated. Thanks in advance.

Logs:
```
[2024-11-11 17:25:01.664][request][INFO] GET /api/config          
[2024-11-11 17:25:01.664][response][INFO] (config) GET /api/config => 200 OK                                                        
[2024-11-11 17:25:04.875][request][INFO] GET /api/devices/knowndevice                                                               
[2024-11-11 17:25:04.877][response][INFO] (get_known_device) GET /api/devices/knowndevice => 200 OK
```

**Reply by BlackDex** [reply]
Are you using a Cloudflare maybe?

**Reply by eltroniq** [reply]
No, im running it only in my intranet. Access from outside only via VPN.

**Reply by BlackDex** [reply]
Then i might think a cipher record which has a key column entry could be causing the issue. But i would have suspected that the key rotation should have solved that actually.

**Reply by eltroniq** [reply]
Thank you for your fast responses and for your time!
Unfortunately I don't know much about this project on the technical side.

I just looked through the database and I am not quite sure what you refer to with a key column entry. Is it even something I can find or filter for in the database?
Or is there another way to find the problematic record?

**Reply by BlackDex** [reply]
The problem is that there is no good way to detect the troubled cipher as far as I know.
I'm not sure if other clients also have issues, or just ignore these specific ciphers since I'm as of yet still unable to reproduce this, and most people who reported issues got it solved by either exporting and importing or deleted new ciphers (which sometimes didn't had the key column filled in the database). So it's still a mystery too me.

You can try to see if you export the vault via two different ways, like web-vault and cli, if there are differences, like one is missing a vault item. That might be helpful.

But other then deleting ciphers (also from the trash) and checking if it works again, i have no clue.
And, we would then love to know what the record was which was causing the issue, to see if we can find something there which might have caused the issue.

**Reply by eltroniq** [reply]
I see. I already got the export from the vault and tried writing a small script to count the entries but think I took a wrong turn there - due to my heavy use of additional and custom fields the CSV is a bit messed up.
Since it's getting late and I don't have my laptop in reach for the CLI, I will get back to you once I have the results in comparing the two exports, hopefully tomorrow.

**Reply by eltroniq** [reply]
I just compared the CSV export between CLI and web vault and unfortunately there are no differences. 
Another thing I discovered is when using the chromium addon, it gives me following error upon login:
`Cannot read properties of null (reading 'kdfType')`
Could this be related?

**Reply by eltroniq** [reply]
I think I finally found the problem!
I always copied my URL over the other URL fields as well when setting the bitwarden client up. Now when only filling the base URL, it works in the extension as well as in the android app.
I can't really explain this behavior but if it works, it works.

Thank you so much for your time and your willingness to help.

**Comment by osorionicolas** [comment]
In my case I didn't have any organization. After create on it started to work properly

**Comment by T0mu** [comment]
I was having issues only on my tablet but not on a freshly reinstalled phone, therefore I inferred it was not a server, or server-client issue.
i started deleting cache and app data. no luck. 
my solution, in the end, was delete and reinstall the bitwarden app.
take note, I had another working bitwarden install on work profile linked to a different server.
even if i deleted only the app in the main profile, on reinstall it worked.
I suppose something got rotten in the profile android folder.
still, I don't really understand why a cache and data clean did not solve the issue.

**Reply by Coding-Kiwi** [reply]
Same for me, I had the newest vaultwarden server version (docker) and the newest android app version.
Uninstalling and reinstalling did the trick.. weird.

Thank you!
